src/hooks/useVenueAccessToken.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from "react";
+import firebase from "firebase/app";
+
+import { AnyVenue } from "types/Firestore";
+
+import { WithId } from "utils/id";
+import { accessTokenKey } from "utils/localStorage";
+import { isTruthy, notEmpty } from "utils/types";
+import { checkAccess, getLocalStorageToken } from "functions/auth";
+
+export const useVenueAccessToken = (
+  venue?: WithId<AnyVenue>,
+  onDenyAccess?: () => void
+) => {
+  useEffect(() => {
+    if (!venue) return;
+
+    const denyAccess = () => {
+      localStorage.removeItem(accessTokenKey(venue.id));
+      onDenyAccess && onDenyAccess();
+    };
+
+    if (notEmpty(venue.access)) {
+      console.log("venue.access", venue.access);
+      const token = getLocalStorageToken(venue.id) ?? undefined;
+      console.log("found token:", token);
+      if (!token) {
+        console.log("checking access");
+        checkAccess({
+          venueId: venue.id,
+          token,
+        }).then((result) => {
+          console.log(
+            "access check result:",
+            result,
+            "isTruthy result.data:",
+            isTruthy(result.data)
+          );
+          if (!isTruthy(result.data)) {
+            firebase
+              .auth()
+              .signOut()
+              .finally(() => {
+                denyAccess();
+              });
+          }
+        });
```

Comment:
If we extract this logic out into a helper function then we can make use of `async`/`await` for the most of it to simplify things.

Reply:
This is outdated, resolving it.